src/Microsoft.Health.Fhir.Shared.Core/Features/Search/BundleFactory.cs

```diff
@@ -65,21 +69,41 @@ public ResourceElement CreateHistoryBundle(SearchResult result)
                     Url = hasVerb ? $"{r.Resource.ResourceTypeName}/{(httpVerb == Bundle.HTTPVerb.POST ? null : r.Resource.ResourceId)}" : null,
                 };
 
-                string statusString;
+                string statusString = null;
+
+                string ConvertStatusString(HttpStatusCode statusCode)
+                {
+                    return $"{(int)statusCode} {statusCode}";
+                }
+
                 switch (httpVerb)
                 {
                     case Bundle.HTTPVerb.POST:
-                        statusString = ((int)HttpStatusCode.Created).ToString() + " " + HttpStatusCode.Created;
+                        statusString = ConvertStatusString(HttpStatusCode.Created);
                         break;
                     case Bundle.HTTPVerb.PUT:
+
+                        if (string.Equals(r.Resource.Version, "1", StringComparison.Ordinal))
+                        {
+                            statusString = ConvertStatusString(HttpStatusCode.Created);
+                            break;
+                        }
+
+                        statusString = ConvertStatusString(HttpStatusCode.OK);
+                        break;
+
                     case Bundle.HTTPVerb.GET:
-                        statusString = ((int)HttpStatusCode.OK).ToString() + " " + HttpStatusCode.OK;
+#if !Stu3
+                    case Bundle.HTTPVerb.PATCH:
+#endif
+                        statusString = ConvertStatusString(HttpStatusCode.OK);
                         break;
                     case Bundle.HTTPVerb.DELETE:
-                        statusString = ((int)HttpStatusCode.NoContent).ToString() + " " + HttpStatusCode.NoContent;
+                        statusString = ConvertStatusString(HttpStatusCode.NoContent);
                         break;
                     default:
-                        throw new NotImplementedException();
+                        _logger.LogWarning(nameof(CreateHistoryBundle) + " tried to resolve a status string for {httpVerb} but a mapping wasn't defined.", httpVerb);
```

Comment:
>_logger.LogWarning(nameof(CreateHistoryBundle) + " tried to resolve a status string for {httpVerb} but a mapping wasn't defined.", httpVerb); [](http://example.com/codeflow?start=24&length=141)

is there are reason why logging preferred to throwing exception?

Reply:
Throwing an exception here will prevent data being returned. Should one unexpected value invalidate the entire payload? I'm open to being convinced otherwise.

Reply:
At least we monitor exceptions, no one looks into warning logs. 
Exception is a pain, but without pain you will never notice system is not working as intended. 

I guess it's a question of how much we care about history not working fully (in a rare case we start support HEAD verb).

Reply:
...makes sense, I'll refactor the tests to catch if new enums are added. I also don't want to randomly find this in prod again